### ISSUE TITLE
Is 18th feature plane required? 

### ISSUE THREAD

**Issue by Neverever** [gcp/leela-zero#687]
## 17th feature plane
From AlphaGo Zero paper:
>  the colour feature C is necessary, because the **_komi_** is not observable

The 17th feature plane is indicating whether komi is given to the opponent. (Black=1, White=0)

## 18th feature plane
The 18th feature plane is added due to

> slight imbalance in that it is easier for the **_white_** player to see the board edge (due to how padding works in neural networks).

AFAIK
the current player's stones are marked as `1` in feature plane 1 to 8
the opponent players stones are marked as `1` in feature plane 9 to 16
There isn't really a concept of black/white players, only myself/opponent

If AlaphaGo Zero's initial design help white player easier to see the board edge, why black is not seeing it? (both players' stones are marked as `1`)

**Comment by FFLaguna** [comment]
I'm also interested in a bit longer of an explanation than I've seen to date on this. I may have missed it somewhere.

**Comment by lightvector** [comment]
When convolutions go past the edge of the board, due to zero-padding it treats it as if the values there were zeros. The presence of a layer that is all ones on the whole board helps the neural net tell what empty spots are on the board versus off the board, which is important for understanding territory and tactics near the edges. Without the 18th feature, only white would have access to such a layer, black would have to implicitly infer the board edge from the fact that nobody has stones there and from the fact that deeper down, convolution channels keep getting "zeroed out" past that edge.

**Comment by Neverever** [comment]
If the Network is rely on the “ones” layer to determine edge of the board
then back in “17 features” days, it’s Black player has the advantage to detmine the edge. 

From AlphaGo Zero paper
> The final feature plane, C, represents the colour to play, and has a constant value of either 1 if black is to play or 0 if white is to play

What/When was the initial observation that White has advantage?
If anyone has issue reference, I’m more than happy to get my hands dirty and dig into this.

**Comment by lightvector** [comment]
Ah, you're right, I got the colors flipped. Yes, it would be black that can see the edge more easily, not white. But the point still stands, either way there will be a weird asymmetry between the colors if you only have 17 and only one of the players gets a plane of 1s.

**Comment by Neverever** [comment]
The 17th plane was designed for Komi indication. 
Whether it’s used by Network internally to determine the board edge? It’s beyond my knowledge. 

The project goal was to replicate AlphaGo Zero
With additional 18th plane, I feel it’s not aligned with the project goal. 

With the board edge issue, I think we should leave it to the Network, it’ll figure out somehow (same for ladder issue)

**Comment by gcp** [comment]
>With additional 18th plane, I feel it’s not aligned with the project goal.

The project goal is to have a go player without human knowledge. A bug that introduces an imblance for black vs white is not aligned with that.

There is no point in replicating clear bugs.

**Comment by Neverever** [comment]
Introducing 18th plane that enables one side to have same knowledge of board edge is clearly a **_human interference_**

Because there is no articles talk about this "bug" on the internet, I would believe the AlphaGo Zero design was fine, it would be more likely somewhere in the system that's causing the bug.

If anyone could provide issue reference or bug description in greater detail, I'm volunteer myself to investigate further.

**Comment by FFLaguna** [comment]
>Introducing 18th plane that enables one side to have same knowledge of board edge is clearly a **_human interference._**

It's not human interference, it's just considered a "[hyperparameter](https://stats.stackexchange.com/questions/309729/what-are-the-real-hyperparameters-of-a-neural-network)". Those are the handful of minor details you can tweak about the neural network before letting the neural network do its blackbox magic in training.
